Services/source/housekeeping/housekeeping_service.c

```diff
@@ -126,21 +130,26 @@ uint16_t get_file_id_from_timestamp(uint32_t timestamp) {
 Result dynamic_timestamp_array_handler(uint16_t num_items) {
   if (num_items == 0) {
     if (timestamps != NULL) {
-      free(timestamps);
+      vPortFree(timestamps);
     }
     hk_timestamp_array_size = 0;
     return SUCCESS;
   }
   if (num_items != hk_timestamp_array_size) {
-    uint32_t *tmp = realloc(timestamps, sizeof(*timestamps * (num_items + 1))); //+1 to allow non-zero index room
-    if (!tmp) { 
+    uint32_t size = sizeof(*timestamps) * (num_items +1); //+1 to allow non-zero index room
+    uint32_t *tmp = (uint32_t*)pvPortMalloc(size);
+    if (tmp == NULL) {
       return FAILURE;
     }
+    if (timestamps != NULL && hk_timestamp_array_size < num_items) { //check if growing because we delete everything if shrinking
+      memcpy(tmp, timestamps, sizeof(*timestamps * hk_timestamp_array_size));
```

Comment:
It looks strange. Does it give the same result as `sizeof(*timestamps) * hk_timestamp_array_size`?

Reply:
Nope it wouldn't. that's similar to a bug I fixed elsewhere here. good catch